lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -510,6 +510,38 @@ def set_information(self, trans, id, payload={}, **kwd):
         trans.log_event('User information added')
         return {'message': 'User information has been saved.'}
 
+    @expose_api
+    def set_favorite_tool(self, trans, id, payload={}):
```

Comment:
I think we should generalize this slightly such that we wont have to add individual endpoints for other `favorite` items e.g. workflows, visualizations etc. by creating an endpoint `set_favorites` instead.

Reply:
Sounds good to me, I can do that.

Reply:
This is now in.